src/ResourceManager/Compute/Commands.Compute/Extension/VmssDiskEncryption/GetAzureVmssDiskEncryptionStatus.cs

```diff
@@ -155,36 +155,98 @@ private PSVmssDiskEncryptionStatusContext GetVmssDiskStatus(string rgName, strin
+                foreach (VirtualMachineScaleSetVM instance in instancesList)
+                {
+                    List<PSVmssDiskEncryptionDiskStatusContext> currentDisks = new List<PSVmssDiskEncryptionDiskStatusContext>();
+                    VirtualMachineScaleSetVMInstanceView vmiv = this.VirtualMachineScaleSetVMsClient.GetInstanceView(rgName, vmssName, instance.InstanceId);
+                    if (vmiv != null && vmiv.Disks != null)
+                    {
+                        foreach (DiskInstanceView div in vmiv.Disks)
+                        {
+                            List<InstanceViewStatus> perDiskEncryptionStatuses = new List<InstanceViewStatus>();
+                            bool isEncrypted = false;
+                            foreach (InstanceViewStatus ivs in div.Statuses)
+                            {
+                                if (ivs != null && ivs.Code != null && ivs.Code.StartsWith("EncryptionState"))
+                                {
+                                    perDiskEncryptionStatuses.Add(ivs);
+                                    if (!psResult.EncryptionEnabled)
+                                    {
+                                        // retains the state of the last status code written for this disk 
+                                        isEncrypted = ivs.Code.Equals("EncryptionState/encrypted");
+                                    }
+                                }
+                            }
+                            // EncryptionEnabled is considered deprecated, and removal would be a breaking change
+                            // This records true for "encryption enabled" if any disk is marked as encrypted 
+                            psResult.EncryptionEnabled |= isEncrypted;
+
+                            // For finer granularity, preserve and add full status for this individual disk 
+                            PSVmssDiskEncryptionDiskStatusContext diskStatus = new PSVmssDiskEncryptionDiskStatusContext
+                            {
+                                Name = div.Name,
+                                EncryptionSettings = div.EncryptionSettings,
+                                Statuses = perDiskEncryptionStatuses
+                            };
+                            currentDisks.Add(diskStatus);
+                        }                 
+                    }
+                    PSVmssDiskEncryptionInstanceStatusContext currentInstance = new PSVmssDiskEncryptionInstanceStatusContext
+                    {
+                        Name = instance.Name,
+                        Id = instance.Id,
+                        Disks = currentDisks
+                    };
+                    psResult.Instances.Add(currentInstance);
+                }
+            }
+            catch (InvalidOperationException)
+            {
+                return psResult;
```

Comment:
why are we returnign a value on error?  Also, why repeat the line below in this block?

Reply:
agreed, copied from prior logic, will clean this up

Reply:
restructured and removed unnecessary line